File: python/benchmark.py
import time
import subprocess
import psutil

if __name__ == "__main__":
    start_time = time.time()

    # Execute your main script (replace 'script.py' with the actual script name)
    result = subprocess.run(['python', 'script.py'], stdout=subprocess.PIPE, stderr=subprocess.DEVNULL, text=True)

    end_time = time.time()
    execution_time = (end_time - start_time) * 1000  # Convert to milliseconds

    print ("Performance of python script file:")
    print(f"Execution Time: {execution_time:.2f} ms")

    # Measure CPU usage and memory consumption
    process = psutil.Process()
    cpu_usage = process.cpu_percent(interval=1)  # Measures CPU usage over a 1-second interval
    memory_info = process.memory_info().rss / (1024 ** 2)  # Memory usage in MB

    print(f"Memory Consumption: {memory_info:.2f} MB")

    print ("---------- END ---------")

File: rust/benchmark.rs
extern crate sysinfo;
use std::time::Instant;
use std::process::{Command, Stdio};
use std::io::Read;
use sysinfo::{System, SystemExt};

fn main() {
    let start_time = Instant::now();

    // Execute your main script (replace 'main' with your actual binary name)
    let mut main_process = Command::new("./target/debug/main")
        .stdout(Stdio::piped())
        .spawn()
        .expect("Failed to execute main.rs");

    let _main_output = if let Some(ref mut stdout) = main_process.stdout {
        let mut output = String::new();
        stdout.read_to_string(&mut output).expect("Failed to read stdout");
        output
    } 
    else {
        String::new()
    };
    
    main_process.wait().expect("Failed to wait for main.rs to finish");


    println!("Performance of rust script file:");

    let end_time = start_time.elapsed();
    println!("Execution Time: {:.4?}", end_time);

    // Initialize a System object for system monitoring
    let sys = System::new_all();

    let memory_info = sys.get_used_memory() / (1024 * 1024); // Memory usage in MB
    println!("Memory Consumption: {:.4} MB", memory_info);

    println!(" ---------- END -----------");
}
